new replication subnet group module

ruimoreira opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
New module for replication subnet group
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->

- New Module Pull Request

##### COMPONENT NAME
dms_replication_subnet_group
##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->

goneri committed:
vmware_datastore_facts: handle external datastore

Without this commit, `vmware_datastore_facts` ignores the datastores are
not associated with a cluster of a datastore. This case can happen if a NFS
share is mounted directly on a ESXi instance.

goneri committed:
vmware_vmkernel: no error on state=absent

The `network.type` parameter is set to default on default on `static`.
As a consequence, the network parameter is always defined as `dict`.

This patch ensures we don't check the value of the network configuration
if we want to delete the interface.

goneri committed:
vmware_vm_facts: fix the support with regular ESXi

Ensure the module still work with the ESXi where CustomFieldsManager
does not exist.

From: https://www.vmware.com/support/developer/converter-sdk/conv60_apireference/vim.CustomFieldsManager.html

    The CustomFieldsManager object is used to add and remove custom fields to
    managed entities.
    The custom fields values set on managed entities are available through the
    customValue property and through the summary objects for VirtualMachine
    and HostSystem. They are not available directly through this managed object.
    This functionality is only available through VirtualCenter.

Fixes: #56071

ruimoreira committed:
new replication subnet group module

ruimoreira committed:
fixed a few linting issues

ruimoreira committed:
fixed a few linting issues

ruimoreira committed:
reduced spacing and fixed indentation issues

ruimoreira committed:
updated docs and examples

ruimoreira committed:
fixed yaml issues with documentation

ruimoreira committed:
fixed pep8 issues

ruimoreira committed:
changed tests to remove static subnet ids, alaso added code to create , VPC, and subnets and destroy them in the end

felixfontein committed:
Hetzner failover IP: refactoring (#56203)

* Rename helper function.

* Extract hetzner.py module_utils.

* Rewrite docs.

* Add module docs fragment.

* Split up get_failover function.

* Add tests for new function.

* hetzner_pass -> hetzner_password

* Move common argspec to module_utils.

chrisvanheuveln committed:
nxos_snmp_user: platform fixes for get_snmp_user (#55832)

* nxos_snmp_user: platform fixes for get_snmp_user

snmp user output behavior varies quite a bit for the different nxos platforms and required several workarounds:

- N5K/N6k
 - These platforms do not support structured output for `show snmp user`.
 - The current code lands in an `except` clause when the output is not structured; so I added a new `get_non_structured_snmp_user` method to scrape the state from the regular cli output if it's present.

- N9K-F
 - The `group` data in the JSON output is different for this platform; it has a different key (just `group` instead of `TABLE_groups` or `group_names`) and it is not indexed
 - For a single group the value is a string, for multiple groups it's a list

- sanity
 - N5K/N6K/N9K-F platforms will reject `no snmp user <name> <role>` when it's the last role defined for the user.
 - workaround is to use `nxos_user` to remove the user

- Changes validated on:
 - `N3K, N3K-F, N35, N6K, N7K, N9K, N9K-F`
 - `6.0(2)A8`
 - `7.0(3)I2, 7.0(3)I4, 7.0(3)I5, 7.0(3)I6, 7.0(3)I7`
 - `7.3(2)D1`
 - `7.3(3)N1, 7.3(4)N1`
 - `8.3(2)`
 - `9.2(2), 9.2(3)`

* fix lint warning

Qalthos committed:
Ensure 'authorize' is always bool, regardless of source (#56271)

Qalthos committed:
eos_l2_interface: Handle `Interface does not exist` (#56140)

* Fail on switchport error

* Handle `Interface does not exist` when gathering switchport.

chrisvanheuveln committed:
nxos_vpc:Fix multiple idempotency issues, add missing commands (#55735)

* nxos_vpc:Fix idempotency issues with multiple attributes

Several attributes were causing idempotency problems on various platforms:

- `auto_recovery`
 - This command can be disabled on certain platforms and will nvgen as `no auto-recovery`
 - When enabled it has an additional optional-keyword for changing the `reload-delay` timer value
  - This was addressed by adding a new attribute `auto_recovery_reload_delay` to handle setting the timer value
  - This new attribute is mutually exclusive with `auto_recovery`

- `/show run vpc/show run vpc all/`
 - Changed the command that gets state to `all` so that it could differentiate between `auto-recovery` and `auto-recovery reload-delay`
 - This change resulted in also changing some attribute handling withing `get_vpc`, since some attributes like `peer_gw` relied on presence of the config to determine state true or false. With `all` the config is always there so these attrs must specifically check for `'no '` in the string.

- `delay_restore`
 - This command has two additional, optional keywords that exist on some platforms and not others.
 - New attrs:
  - `delay_restore_interface_vlan`
  - `delay_restore_orphan_port`

- Modified the `sanity` test to include the new attributes and to fix the platform issues.

- Bugfix Pull Request

`modules/network/nxos/nxos_vpc.py`

- Validated `nxos_vpc` `sanity` test on these platforms, all are now 100% Pass: N35, N3K, N3K-F, N6K, N7K, N9K, N9K-F

- TBD: Future work is needed to add support for `peer_gw_exclude_gw` timers. This could be addressed in the same way as the `auto_recovery_reload_delay` changes included here.

* lint fix

* Add 'version_added' tags for new options

justjais committed:
To fix IOS static route CI failure (#56292)

* ios static failure

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* fix CI failure

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

acozine committed:
should have gone into 52373 (#56306)

Commit:
NetApp ONTAP module for manage ipspaces (#49821)

* NetApp ONTAP module for manage ipspaces

* fixes for ci_cd and code layout

* indentation fixes

* code style fixes

* fixing yamllint issue

* unit test for module na_ontap_ipspace

* fixing sanity tests

* change pytest.skip to pytest.mark.skip

* adding ansible version to 2.9

akshay1610 committed:
Update os_image documentation (#49705)

* Update os_image.py docs to show auth details

Akasurde committed:
VMware: support folder param in vmware_vm_facts (#56298)

folder can be used as a filter while gathering facts about VMs.

Fixes: #56125

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: vmware_guest_disk support for use_instance_uuid (#56107)

All vmware_guest_* module support for use_instance_uuid from Ansible 2.8,
somehow vmware_guest_disk missed from this change.
This fix adds support for use_instance_uuid in vmware_guest_disk.

Fixes: #56021

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

anasbadaha committed:
Adding Support for Traffic Class in Onyx Switches (#55577)

* Adding Support for Traffic Class in Onyx Switches

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Enhancing the code and elemenating code duplicate

Signed-off-by: Anas Badaha <anasb@mellanox.com>

anasbadaha committed:
Adding Support for NVE Protocol in onyx_protocol (#55513)

* Adding Support for NVE Protocol in onyx_protocol

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Pep8 Failures in onyx_protocol.py

Signed-off-by: Anas Badaha <anasb@mellanox.com>

anasbadaha committed:
Adding Support For EVPN in BGP Module (#55503)

* Adding Support For EVPN in BGP Module

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Pep8 Failures in onyx_bgp.py

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Pep8 Failures in onyx_bgp.py Phase 2

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Samer's Comments

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Remove file Variable

Signed-off-by: Anas Badaha <anasb@mellanox.com>

felixfontein committed:
Hetzner failover IP info module (#56186)

* Add hetzner_failover_ip_info module.

* Extend return value.

* Adjust to changes from refactoring.

* Forgot to adjust example.

RenderQwerty committed:
inventory: virtualbox: fix typo in examples (#56328)

Typo fix in 'windows' word

billdodd committed:
redfish: Add If-Match ETag header to PATCH requests (#56150)

* add If-Match ETag header to PATCH requests

* add changelog fragment

* ensure header keys are lowercase

* fix dict comprehension for python 2.6 and earlier

maiqueb committed:
Added the mtu property to the os_network module (#55881) (#54730)

The networking API v2 specification, which is implemented
by openstack neutron, features an optional MTU parameter that
allows operators to specify the value for the maximum
transmission unit value.

ruimoreira committed:
adding missing variable

ruimoreira committed:
adding missing variable

Fred-sun committed:
modify azure_rm_aks* document (#56293)

ashleyconnor committed:
Fix typo in aws log group return documentation (#56335)

Change from objetcs to objects

chrisvanheuveln committed:
nxos: remove deprecated test target files (#55949)

* nxos_switchport: remove deprecated test target files

* Remove tests for nxos_ip_interface, nxos_portchannel; update nxos_interface

- `nxos_ip_interface` and `nxos_portchannnel` tests have been removed

- `nxos_interface` now uses current module `nxos_l3_interface` instead of `nxos_ip_interface`

krisvasudevan committed:
Bug fixes for the nios member object (#56045)

* Bug fixes in api.py

* Examples updated in nios_member.py

* Update lib/ansible/module_utils/net_tools/nios/api.py

Co-Authored-By: Sumit Jaiswal <sjaiswal@redhat.com>

justjais committed:
To fix the issue where nios_zone module was running second time for same object (#55595)

* fix for 55128 issue

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* review comment fix

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

Fred-sun committed:
update azure_rm_applicationsecuritygroup document (#56344)

samdoran committed:
Update default roles_path documentation (#56320)

sbonds committed:
ansible-config doesn't edit configs (#56355)

Avoid suggesting that it does in the description since that capability doesn't yet exist.

tzurE committed:
IBM_Storage: fixed a bug - Unwanted arguments cause failure (#56357)

* IBM_Storage: fixed a bug which caused several of our modules to pass unwanted arguments and fail

* IBM_Storage: Added a changelog for 56357 bug

mattclay committed:
Add sanity test to check for unwanted files.

isaac88 committed:
docs: Update scopes value example (#56362)

Existing example in GCE guide returns "invalid scope" error; value needs "https://" prefix

felixfontein committed:
Add myself to committer list. (#56363)

mattclay committed:
Revert oracle (#56364)

* Revert "Update vcn module documentation (#55888)"

This reverts commit 94ef24974a053c7eeee5ec3c4bd3f0aba00712e0.

* Revert "Initial commit for Oracle Cloud Infrastructure modules (#53156)"

This reverts commit 07b9f52fd5670ff85c2a36028e3d476882eaec83.

* Revert "Add Oracle module maintainers (#55889)"

This reverts commit b0f3df360aa0f99bd3878c844730f7c124c397d7.

goneri committed:
vmware: use hostname in esxi_hostname

The use of the `hostvars[esxi1].ansible_host` was attempt to use
different hostname and IP address. But it's actually the source of
more problems.

VMware expects to be able to resolvable the host name. This means, that
if someone wants to run the test-suite, s/he needs to use a DNS or
update the `/etc/hosts` files on the different hosts.

goneri committed:
vmware_vm_facts: fix the test with vcsim

The VM is call `ha-host_VM0`, not `DC0_H0_VM0`. This commit adjusts
the test to make it more resilient if the VM name is different.

Rahul-CSI committed:
Fixed snapshot module, Resource not found issue for persistent-disk-0 (#35341)

resmo committed:
Revert "inventory: virtualbox: fix typo in examples (#56328)" (#56374)

This reverts commit 099917ddc8744fa1fb864600c966b69901778b7b.

Akasurde committed:
VMware: Improve vmware_portgroup (#56382)

* Rewrite vmware_portgroup module
* support check mode
* VLAN ID isn't required anymore
* VLAN ID 0 (no tagging) is the default
* Options match values in vSphere Client and vmware_vswitch module
* Policy override is configured properly
* VMware: vmware_portgroup updates

Fred-sun committed:
update azure_rm_autoscale document (#56388)

Commit:
Issue55222 (#55223)

* Python 3.7 compatibility for re.Pattern (Fixes #55222)

* Python 3.7 compatibility for re.Pattern (Fixes #55222)

samccann committed:
Fix 2.8 changlog url and reverse order porting guides TOC (#56373)

* fixed url for 2.8 changelog, fixed spelling issues, reversed TOC order

NilashishC committed:
Fix VyOS cli prompt issues (#56389)

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

pilou- committed:
[doc] command module: mention all parameter types (#56396)

bcoca committed:
fix module defaults (#56020)

* fix module defaults

 - corrected precedence (specific module > group)
 - made into reusable function
 - use from gather_facts/service/package to match 'actual module used'

pilou- committed:
[doc] fix example: always trigger the handler (#56384)

by default, debug task result isn't changed - add 'changed_when: yes' to trigger handler

pilou- committed:
[doc] expect module: mention all parameter types (#56394)

Fred-sun committed:
update azure_rm_appgateway document (#56339)

* update azure_rm_appgateway documentation

Fred-sun committed:
update azure_rm_appserviceplan related module documents  (#56348)

* update azure_rm_appserviceplan module documentation

wltjr committed:
Modify/fix handling of noreplace option issue #55365 (#55967)

Presently setting noreplace: no/false has no effect. Negating the default
to true from false and modifying conditionals seems to correct this.
Making it such that you can re-install and existing package. That did not
seem possible without such modifications.

Potential fix for issue #55365 pending issue reporter testing and feedback.

nalsaber committed:
Initial commit - Oracle modules

Revert "Revert oracle (#56364)"

This reverts commit eefc90b66f7d02338a1484ce6a1350eddf640d05.

LKaemmerling committed:
Fix hcloud tests

nalsaber committed:
Initial commit for Oracle Cloud Infrastructure modules

mattclay committed:
Remove old references to `cloud-config-aws.yml`.

References to the new `cloud-config-aws.ini` file are provided by `"$@"` in tests.

Tomorrow9 committed:
Merge pull request #56383 from Tomorrow9/fix_pr56296

VMware: add timezone and hwclockUTC customize parameters for Linux guest in vmware_guest

jeking3 committed:
azure_rm_virtualmachine (and _facts): add boot_diagnostics control and facts (#49661)

jctanner committed:
Nullify improperly licensed test file

mkrizek committed:
dnf: fix wildcard matching for state: absent (#56013)

* dnf: fix wildcard matching for state: absent

Fixes #55938

* Add changelog...

* Fix sanity check failure...

jborean93 committed:
winrm - fix failure parsing for send input errors (#56245)

ShachafGoldstein committed:
Change S3 bucket urls to new method as per #56124 (#56418)

* Change S3 bucket urls to new method as per #56124

* duplicate / and dded changelog fragment

* Revert "duplicate / and dded changelog fragment"

This reverts commit b89d5cbe6d983168212c44def537d87f0ed22204.

* duplicate /

jeking3 committed:
azure_rm_securitygroup - fix delete error handling (#56422)

pratikgadiya12 committed:
VMware: Check to retrieve list of all used disks (#56380)

Fixes: #56290

Akasurde committed:
Fixed error handling in github_issue module (#39652)

* Fixed error handling in github_issue module

Due to recent changes in github3.py library module stopped working.
This fix adds extra error handling for new changes in library.

Fixes: #39627

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

* Check version

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

* Refactor github_issue

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

maxamillion committed:
YUM - handle enable of non-existent repo (#53286)

ciliusmp committed:
hetzner changed image tag for server, so its working with ids and names (#56406)

* hetzner changed image tag for server, so its working with ids and names

* changed trailing whitespace

* added higher retries count since server creation from own image takes a lot longer

Akasurde committed:
VMware: fix inventory plugin (#56431)

* Fixed regression introduced in #56071
* Added test to check if hostvars are populated
* Fix toml installation logic

Fixes: #56413

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

netsiphon committed:
Allow config commands in aireos_command to support prompts (#56130)

* Switch failure on config command to just a warning.

* Change wording in documentation

* Missing whitespace

sc68cal committed:
[eos] Allow ports to be configured with just trunk mode (#55887)

Fixes #55886

pabelanger committed:
Remove hardcoded ansible user for junos_user integration tests (#56452)

It is possible we are using a different username then 'ansible' for
testing junos. By using ansible_user_id, this should be a more dynamic
check.

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

kbreit committed:
Meraki - Add check for auth_key parameter (#56199)

* auth_key parameter is required
- This will have to change when httpapi is implemented

* Add integration test

* Add assertion

* Enable VLANs on network

* Add required for auth_key

m3adow committed:
Small typo fix in documentation (#56480)

The documentation was probably C&P'd from user, leaving a "useradd" mention where a "groupadd" should be.

chashtag committed:
Update ovirt_vnic_profile.py (#56466)

##### SUMMARY
Update documentation to reflect module name


##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

thomasboussekey committed:
Update of Ansible 2.8 roadmap (add RC2 & RC3) (#56436)

##### SUMMARY

* Remove **if needed** for RC2 & RC3
* Set correct release date for RC3

samccann committed:
update release table for 2.8 (#56486)

samccann committed:
changed version to 2.8 (#56487)

bcoca committed:
removed previouslly deprecated settings (#55662)

* removed previouslly deprecated settings

  fixes #55304

* also removed from play context

goneri committed:
vmware_host_datastore: ability to target a ESXi

Until now, the module was only able to interact with vcenter. This
commit adds the ability to directly target an ESXi without the
`esxi_hostname` parameter.

samccann committed:
[Doc-Release-2.8] fixed broken module links in 2.8 porting guide (#56494)

* fixed broken module links

acozine committed:
gcp_compute_target_proxy does not exist, update deprecated redirect (#56496)

spidermila committed:
VMware: return info about SCSI controller in vmware_guest_disk_facts (#56442)

vmware_guest_disk_facts module added bus number of the SCSI controller to the output

goneri committed:
VMware: vsphere_copy: ability to target an esxi instance (#55930)

*`vsphere_copy` was only able to interact with a vCenter instance. This
patch change that.
* In addition, it also makes use of the `vmware_argument_spec`.

Co-Authored-By: Abhijeet Kasurde <akasurde@redhat.com>

AravindBalajiS committed:
VMware: Return facts about rule_affinity vmware_drs_rule_facts (#55813)

Added the rule_affinity while returning about the rule facts

davidmhewitt committed:
VMware: Add new module vmware_folder_info (#54344)

* Adds a new vmware module to support getting the folders and their paths within a datacenter
* Add integration tests
* Bump version added
* Refactor integration test
* Improve grammar in docs

Andersson007 committed:
postgresql_db: added tablespace support (#56390)

goneri committed:
vmware: test-refactoring

- In order to keep the integration with `ansible-test`, we prefer to avoid any
  interaction with the Ansible inventory file.
- split up the prepare_vmware_tests/defaults/main.yml in two
  configuration files: one for vcsim and one for a real environment
- remove all the access to hostvars
- directly interact with the ESXi to mount/umount the datastore
  https://github.com/ansible/ansible/pull/56516
- record the virtual machine folder in the environment configuration
- vmware_guest_move: Use https://github.com/ansible/ansible/pull/55237

acozine committed:
Cloud dev docs to rst (#56485)

* Moves developer docs for AWS, ovirt, and openstack modules out of lib/ansible/, integrates them with dev_guide, with abadger's fix to make python snippets pass rstcheck

Fred-sun committed:
Update azure_rm_cdnprofile related module document (#56503)

* modify azure_rm_cdnprofile module documentation

goneri committed:
VMware: vmware_datastore_maintenancemode: check params (#56515)

Check the `datastore` parameter and raise an error if nothing has been
found.

acozine committed:
adds 2.9 roadmap, reversed local TOC order (#56371)

Qalthos committed:
Properly handle check for session support on eos (#56407)

* Properly handle eos session check.

* Why is eos_use_sessions an int?

* Add changelog

sivel committed:
Bump erlang ping to 1:20.3.8.18-1

Qalthos committed:
Fix eos_support_session logic (#56524)

kbreit committed:
meraki_switchport - Improve reliability (#54275)

* Rewrite much of the execution of meraki_switchport
- Previous versions had problems with idempotency and allowed_vlans

* Modified payload creation
- Parameter map is used
- propsed is created using .copy()
- Much cleaner this way

* Add whitespace for lint

* Add bugfix snippet for changelog

mattclay committed:
Temporary fix for vsphere_copy CI failure.

s-hamann committed:
Add seed parameter to random_mac filter (#51841)

chashtag committed:
Update ovirt_vnic_profile.py (#56518)

##### SUMMARY
qos is not defined when port_mirroring is enabled, a NoneType is returned for entity.qos in this case
getattr is safer to use than using a direct call to entity.qos.name


##### ISSUE TYPE
- Bugfix Pull Request

+label: docsite_pr

pilou- committed:
consul modules: update documentation (#56408)

* consul modules: Python 2.6 is always required on managed node, document all types, improve parameter descriptions, fix typos
* consul_kv: add doc for retrieve parameter

sivel committed:
When not using file_per_task, make sure we don't prematurely close the perf files (#56477)

* When not using file_per_task, make sure we don't prematurely close the perf files

* add changelog fragment

goneri committed:
vsphere_copy: fix the E309 error (#56537)

Resolve the two following errors:

ERROR: lib/ansible/modules/cloud/vmware/vsphere_copy.py:0:0: E309 version_added for new option (host) should be '2.9'. Currently None
ERROR: lib/ansible/modules/cloud/vmware/vsphere_copy.py:0:0: E309 version_added for new option (login) should be '2.9'. Currently None

mjmayer committed:
Aws waf region (#48953)

* Add waiter for AWSRegional

* Add support for WAF Regional

* Add support for regional waf web acl

* Remove set_trace, pep formatting

* Add paginator for regional_waf

* Change name of param for waf_regional

This is more in line with how AWS refers to the service. Additional
 changes made to how client is called. Used ternary to reduce if
 statements

* Change parameter name to waf_regional

* Add support for removal waf regional condition

* Change parameter from cloudfront to waf_regional

* Added state: absent waf rule

* Remove set_trace

* Add integration tests for waf regional

* WIP: adding region parameter to tests

* Add support for waf facts module

* Add region to waf regional integration tests

* Update security policy for waf regional testing

* Add type to documentation for waf_regional param

samccann committed:
move too old version to 2.3 (#56540)

kbreit committed:
meraki - Unchanged requests now return the original data (#53576)

* Unchanged requests now return the original data

* Add changelog fragment

* Add integration tests for returned data for meraki_network

* Add integration tests for returned data

* Improve idempoetent output
- Make MX l3 rules always show default rule
- Add integration tests

* Add integration tests for returned data on meraki_network

* Improved idempotency in a few modules and improved tests

yuwzho committed:
Fix the test syntax (#56557)

rohitthakur2590 committed:
asa_config fix (#56559)

Signed-off-by: Rohit Thakur <rohitthakur2590@outlook.com>

Fred-sun committed:
update azure_rm_containerregisty document (#56560)

pmarques committed:
Add os_profile to VM facts, allow to filter by OS (#48040)

Fred-sun committed:
Update azure_rm_containerinstance related module document (#56556)

Fred-sun committed:
Update azure_rm_cdnendpoint related module document (#56440)

mkrizek committed:
register: fail when invalid var name is specified (#56456)

R1AACR committed:
Fix typo in mount doc (#56566)

resmo commented on lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py:
```diff
@@ -0,0 +1,248 @@
+        description:
+            - State of the subnet group
+        default: present
+        choices: ['present', 'absent']
```
not a blocker but please add `type: <...>` for all params, it will be sanity checked in the future.

resmo commented on lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py:
```diff
@@ -0,0 +1,248 @@
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
```
please use short notation of the GPL header

ruimoreira replied:
same

resmo commented on lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py:
```diff
@@ -0,0 +1,248 @@
+    global module
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        required_if=[],
```
can be removed

ruimoreira replied:
same here

resmo commented on lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py:
```diff
@@ -0,0 +1,248 @@
+    exit_message = None
+    changed = False
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
```
there is a helper "missing dependency" error output. please see exsiting modules

ruimoreira replied:
can you please point me in the right direction here ?

resmo commented on lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py:
```diff
@@ -0,0 +1,248 @@
+        if replication_subnet_exists(subnet_group):
+            if compare_params(subnet_group["ReplicationSubnetGroups"][0]):
+                changed = True
+                exit_message = modify_replication_subnet_group(dmsclient)
```
why not implement check mode?

~~~python
if not module.check_mode:
    exit_message = modify_replication_subnet_group(dmsclient)
else: 
    exit_message = " a phrase that makes sense in check mode"
~~~

ruimoreira replied:
also addressed

kbreit committed:
meraki_ssid - Add examples to documentation (#56425)

* Add standalone copy/paste-able examples

resmo commented on lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py:
```diff
@@ -0,0 +1,248 @@
+def main():
+    argument_spec = dict(
+        state=dict(choices=['present', 'absent'], default='present'),
+        subnetgroupidentifier=dict(required=True),
```
not a blocker but I find these long params very hard to read and because the module is named `dms_replication_subnet_group`, why not just skip `subnetgroup` and only use `identifier` and `description` ?

ruimoreira replied:
@resmo this has been addressed

Commit:
Fix typo that breaks invocation of os_stack (#56575)

* Fix typo that breaks invocation of os_stack

* Apply tags conditionally so that the module does not throw up an error when using an older distro of openstacksdk

Qalthos committed:
Fix eos_l2_interface idempotency (#56531)

* Attempt to handle just mode trunk properly

* Add test for trunk-only config and clean up tests

* Add missing eapi tests and remove references to provider as we do not test local

chaitanyaavi committed:
Updated utils to remove Avi SDK dependency and Avi 18.2.2 version update (#54894)

* Updated utils to remove Avi SDK dependency and Avi 18.2.2 version update

* Fixed the python 3.x errors failing for avi_disable_session_cache_as_fact not properly documented

* Updated version added fields for new parameters

* fixed pep8 errors

* made requests import optional

* removed setting requests to None

* Added try catch for the avi helper methods such that any import fails then module fail gracefully. This was needed to pass the requests library not found error

* removed deprecated modules. Also, trying another fix to deal with requests import error

* Fixed python3 errors

* fixed pep8, no-dict-iteritems and import test failures

* added version 2.8 for new field

* some more code cleanup and formatting

* updated the fail message and fixed plint errors

* added workaround for unicode pylint

* fixed the version added for new parameter app_learning_memory_percent and removed unicode_literals import

* Removed check of HAS_AVI for common argument spec

* Updated version added value from 2.8 to 2.9

* Version added value fixes of CI error

sivel committed:
Whitelist listen as a valid keyword on TaskInclude (#56586)

* Whitelist listen as a valid keyword on TaskInclude. Fixes #56580

* Move 'listen' to HandlerTaskInclude

* Remove trailing newline

mattclay committed:
Remove unused cloud config files.

These files have no associated cloud config plugin for ansible-test.

anasbadaha committed:
Adding New Model onyx_qos for Configuring QoS on Onyx Switches (#55127)

* Adding New Model onyx_qos for Configuring QoS on Onyx Switches

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Pep8 Failures in onyx_qos

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Pep8 Failures phase 2

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Samer's Comments on onyx_qos Module

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Shippable Comments Phase 3

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Current Version 2.9

Signed-off-by: Anas Badaha <anasb@mellanox.com>

sivel committed:
Add compat tojson filter for jinja2 versions missing it (#56596)

kbreit committed:
Move backport guide to 2.8 (#56578)

The backport guide should refer to Ansible 2.8 instead of 2.7 as the target branch, since 2.8 is out.

Andersson007 committed:
lib/ansible/parsing/ajson.py: added UNIT-tests (#56398)

Andersson007 committed:
module_utils.postgres: added unittests (#56381)

monkz committed:
Fix typo in pg server ping (#56608)

evgenyfedoruk committed:
Changing the license to Apache 2

felixfontein committed:
docker connection plugin: make privilege escalation code more similar to local.py (#56288)

* Make more similar to local.py

* Fix typo.

ruimoreira committed:
- shorter parameter names
- removed unused dict for requried_if
- shorter GPL Header

ruimoreira committed:
- shorter parameter names
- removed unused dict for requried_if
- shorter GPL Header

yuwzho committed:
Add service endpoints when creation (#56561)

rajaspachipulusu17 committed:
Pluribus Network vrouter ospf module with unit tests (#56435)

* Pluribus Network vrouter ospf module with unit tests

* Sanity fix

* Doc fixes

rajaspachipulusu17 committed:
Pluribus Networks vrouter loopback interface module with UT (#56450)

* Pluribus Networks vrouter loopback interface module with UT

* Sanity fixes

evertmulder committed:
BUGFIX updating existing managed filters fails (#56631)

mkrizek committed:
Templar: encapsulate _available_variables (#55435)

Ensure variables are reset between iterations

Commit:
azure_rm_azurefirewall (#56359)

* azure_rm_azurefirewall

* a few updates

* fixed mistake

* updated docs, fixed syntax

* updated priority

* use checkmode for idempotence test, as it's very long

* fix check mode

modular-magician committed:
Bug fixes for GCP modules (#55977)

modular-magician committed:
Bug fixes for GCP modules (#55975)

modular-magician committed:
Bug fixes for GCP modules (#55974)

modular-magician committed:
Bug fixes for GCP modules (#55973)

felixfontein committed:
Clarify documentation on _info vs. _facts modules (#56611)

* Clarify documentation on _info vs. _facts modules.

* Expand with examples for facts and non-facts.

fernandezcuesta committed:
Update gcp_compute_instance_facts.py (#56581)

swapped dot inside brackets leading to wrong url

Ninroot committed:
Update vagrant.py (#56568)

`inventory_file` has been renamed to `inventory_path`.

resmo committed:
vultr: streamline docs (#56563)

Fred-sun committed:
update azure_rm_availabilityset related module document (#56432)

* Update azure_rm_availabilityset.py

JonTheNiceGuy committed:
Update uri.py (#56395)

1. Note that uri doesn't honor the no_proxy environment variable (due to https://github.com/ansible/ansible/issues/52705), and suggest a work around.
2. Added an example showing a test waiting for a URL to become available (using the `until:`, `retries:` and `delay:` settings) - based on https://gist.github.com/mikeifomin/67e233cd461331de16707ef59a07e372#gistcomment-2718587

Co-Authored-By: Felix Fontein <felix@fontein.de>

mirumu committed:
Update unarchive.py (#56285)

Clarify that the unarchive module does not unpack a compressed file, only archive files.

sethhochberg committed:
Replace deprecated ansible-container references (#56427)

Link to ansible-bender instead of the deprecated ansible-container.

Co-Authored-By: Felix Fontein <felix@fontein.de>

sivel committed:
Don't rely on netloc for determining hostname and port, just use hostname and port (#56270)

* Add changelog fragment
* Fix IPv6 address parsing for py2.6, and add tests
* make sure hostname isn't None

mhoroschun committed:
iam_role.py remove_policies should remove all of the requested policies (not just the first) (#56331)

The remove_policies function in iam_role.py enumerates a list of policies to remove. However, due to an indentation issue on the return True line, only the first such policy would be removed.

This change outdents the return True so that all of the the requested policies are removed.

willthames committed:
Allow python_requirements_facts to cope with packages with dashes (#56166)

* Allow python_requirements_facts to cope with packages with dashes

```
python_requirements_facts:
  dependencies:
    - kubernetes-validate
```

should work as expected

* Ensure tests run for python_requirements_facts

das7pad committed:
[docker] support the lookup of images by digest (#56649)

* [docker] images: add support for lookup by sha256 digest

Signed-off-by: Jakob Ackermann <das7pad@outlook.com>

* [tests] docker image by digest: work on a minimal test case

Signed-off-by: Jakob Ackermann <das7pad@outlook.com>

* [docker] group branch conditions per lookup

Co-Authored-By: Felix Fontein <felix@fontein.de>

* [misc] add a news fragment for the added digest lookup for docker images

Signed-off-by: Jakob Ackermann <das7pad@outlook.com>

machacekondra committed:
kubevirt_vm: Improve create VM from template (#55927)

Zeophlite committed:
Move update_cache to after self.yum_baseurl definition (#56613)

Akasurde committed:
VMware: Add folder param in vmware_datastore_cluster

* Make datacenter as alias and optional
* Add folder param to place datastore cluster in specific folder
* Updated examples
* Updated tests

Fixes: #48010

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
changed type for strings

Commit:
changed type for strings

cinek810 committed:
Raise error in case of empty hosts list in playbook (#56354)

Fixes cases of playbook contaiing:
- hosts:
  -
ERROR! Unexpected Exception, this is probably a bug: sequence item 0: expected string, NoneType found

jonpugh committed:
Out put the version of the dependency so we can better understand the problem. (#56379)

mkrizek committed:
yum: use Custom Group in integration tests (#56439)

mkrizek committed:
yum: fix false error msg about autoremove support (#56459)

* yum: fix false error msg about autoremove support

samccann committed:
Add links to quickstart guides (#56722)

* shell of quickstart

* link to existing quickstart guides

Commit:
na_ontap_volume - added qos policy module options (#56026)

* added qos policy module options

* reverting reformatting not needed

* removing unneeded variable

* updating docs and examples

* adding example with adaptive qos policy

Fred-sun committed:
remove quote (#56690)

samccann committed:
fixed doc build errors (#56735)

Fred-sun committed:
update more azure documentation (#56700)

Fred-sun committed:
Update azure_rm_devtestlab related module document (#56681)

* update azure_rm_devtestlab documention

Fred-sun committed:
Update azure_rm_deployment related document (#56627)

* add azure_rm_deployment documentation

Commit:
Adds one more example for `rollbar_deployment` module (#56535)

- Also `name` added to existing example in order to comply documentation styleguide

felixfontein committed:
Clarify that GPLv3+ module_utils need core team approval (#56740)

* Clarify that GPLv3+ module_utils need core team approval.

* Update docs/docsite/rst/dev_guide/shared_snippets/licensing.txt

Co-Authored-By: Sam Doran <sdoran@redhat.com>

* More 'module_util' -> 'file in module_utils'.

jimwitte committed:
Update get_certificate.py (#56594)

* Update get_certificate.py with an example to calculate number of days until cert expires from get_certificate result.

* Update lib/ansible/modules/crypto/get_certificate.py

Co-Authored-By: Felix Fontein <felix@fontein.de>

csillab committed:
[WIP] Tip to see the content of a single encrypted var (#37887)

* Tip to see the content of a single var encrypted with Ansible Vault

ksubileau committed:
win_file_version: fix typos (#56751)

##### SUMMARY
Minor fixes in error messages


##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
win_file_version

##### ADDITIONAL INFORMATION

jscaltreto committed:
lookup: consul_kv: Add datacenter parameter (#56647)

willthames committed:
k8s minor bug fixes (#56168)

Ensure `wait_condition`s with `Status: Unknown` actually
complete

Return k8s object after wait rather than k8s object before
wait when object is patched.

Akasurde committed:
VMware: Add folder option for vmware_dvswitch (#56707)

* Added folder param
* Mark datacenter as optional parameter
* Updated docs
* Updated tests

Fixes: #54986

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Fred-sun committed:
remove azure_rm_cdnprofile quote (#56695)

Andersson007 committed:
postgresql_db: bugfix 56703 (#56734)

swartzn committed:
Fix workload profileId indexing in netapp_e_volume module (#56747)

dagwieers committed:
Revert "ACI: Implement encoded query_string" (#56783)

This reverts commit 1690af04fe42c657cbf0092d18c86bb517fe86a7.

bcoca committed:
fix collection adjacent loading with --playbook-dir (#56194)

* added inventory adjacent test

bcoca committed:
preserve underlying block building error (#56465)

fixes #56462

s-hertel committed:
The InventoryData class does not have a get_vars method and InventoryManager's get_vars never appears to have been used (#56804)

yungezz committed:
add azure_rm_keyvault_info (#55497)

Deathice committed:
Adding example for add host to groups (#56756)

updates the documentation for the virtualbox plugin

Fred-sun committed:
modify document (#56762)

Fred-sun committed:
modify azure_rm_devtestlabcustomimage's document (#56763)

Fred-sun committed:
Update azure_rm_cosmosdbaccount related module document (#56624)

* update azure_rm_cosmosdbaccount_facts document

Fred-sun committed:
modify azure_rm_devtestlabenvironment relate document (#56769)

Fred-sun committed:
modify azure_rm_devtestlabpolicy relate document (#56770)

sivel committed:
Improvement on setting VALID_INCLUDE_KEYWORDS on HandlerTaskInclude (#56801)

paddynewman committed:
Add a new "log_folder" option to the log_plays callback plugin. (#56717)

* Add a new "log_folder" option to the log_plays callback plugin.

A a new option for the log_plays callback plugin which allows the user to
control where the callback where creates log files.

The option can be set via the ANSIBLE_LOG_FOLDER environment variable or in the
Ansible configuration file, e.g.:

[callback_log_plays]
log_folder = /path/to/my/log/folder

bcoca committed:
added missing version_added

felixfontein committed:
openssl_pkcs12: fix crash due to wrong path used for loading key in check mode (#56808)

* Used wrong path for private key.

* Add changelog.

RobW3LGA committed:
construct_deep_url() (#53475)

* Updates aci.py with the ability to add ACI objects to any depth

Changes start at line 411 (construct_deep_url() and supporting functions). One minor change to line 633 (the original construct_url()) to provide for testability:  ...join(sorted(self.child_classes)) vs ...join(self.child_classes)

I am also attaching two test files. One characterizing the existing construct_url() and the matching test set for construct_deep_url() to support my efforts and proof of parity

* Two PyTest files to support construct_deep_url

These two files provide testing parity, one characterizing the original construct_url() function and the other proofing construct_deep_url(). The ...deep_url.py test file goes five layers deep to provide better validation for the function

* Correcting previous upload to incorrect folder

These two files provide testing parity, one characterizing the original construct_url() function and the other proofing construct_deep_url(). The ...deep_url.py test file goes five layers deep to provide better validation for the function

* Deleting for file name change per Matt Clay

* Deleting for file name change per Matt Clay

* Correcting file names per Matt Clay

@mattclay Thanks again for your continued guidance and patience. Please cancel the previous (incorrect) request

* Wrong location for test file

* Wrong location for test file

* First attempt to comply with suggestions

lib/ansible/module_utils/network/aci/aci.py:517:0: SyntaxWarning: "is not" with a literal. Did you mean "!="?
lib/ansible/module_utils/network/aci/aci.py:534:0: SyntaxWarning: "is not" with a literal. Did you mean "!="?
lib/ansible/module_utils/network/aci/aci.py:558:161: E501 line too long (210 > 160 characters)

* First attempt to comply with suggestions

test/units/module_utils/network/aci/test_aci_construct_url.py:1:14: SyntaxError: import pytest
test/units/module_utils/network/aci/test_aci_deep_url.py:1:14: SyntaxError: import pytest
test/units/module_utils/network/aci/test_aci_construct_url.py:0:0: use "\n" for line endings instead of "\r\n"
test/units/module_utils/network/aci/test_aci_deep_url.py:0:0: use "\n" for line endings instead of "\r\n"
Shortened test function names (less descriptive)

* Second attempt to comply with suggestions

* Second attempt to comply with suggestions

* Third attempt to comply with suggestions

* Third attempt to comply with suggestions

* Pro Tip: Convert from 'CRLF' to 'LF' in VSCode

It's on the status bar to the right

* Added setup() support for tests

* Continued corrections to support testing

* Added two mocks to support testing

I could not find where to place fakes/mocks, so please let me know if the current location is incorrect

* Adding tmpdir property to mock_basic.py

* Added last blank line to mock_basic.py

To pass sanity test

* Attempt to correct setup() issues

* Attempt to correct setup() issues

* Attempt to correct setup() issues

* Attempt to correct setup() issues

* Withdrawing pending injectability tweak to aci.py

* Withdrawing pending injectability tweak to aci.py

* Withdrawing pending injectability tweak to aci.py

* Withdrawing pending injectability tweak to aci.py

tknipper11 committed:
ACI CDP Interface Policy Module Creation (#54818)

* initial commit

* Fix typo in integration test which caused an assertion to fail

* Updated integration testing

* Fix Ansible Sanity Errors

* Changed version added to 2.8 from 2.9

* added from __future__ import absolute_import, division, print_function
__metaclass__ = type

mmyyrroonn committed:
Show access keys for azure_rm_storageaccount (#56612)

jborean93 committed:
win_get_url: ignore defender false positive in tests (#56812)

JulienPalard committed:
docs: cron: fix for example raises a warning. (#56701)

The following one:

    The value 2 (type int) in a string field was converted to '2'

billdodd committed:
redfish: refactor service_root variable (#56198)

goneri committed:
VMware: vcenter_folder: print full path of the new folder (#55237)

Print the full path of the freshly created folder in the output string.

Qalthos committed:
eos_l2_interface handle "Interface does not exist" (#56787)

* Quick and dirty attempt to handle eapi error

* Well this should probably change

* Hopefully this works correctly?

* Fix check_rc handling with httpapi

* Add tests that should hopefully cover the error

* Fix warnings

* Improve tests

bcoca committed:
Prevent vars premature templating (#56117)

Avoid premature vars templating
  * added tests
  * avoid 'is template' warning in vars, since we want them for latter templating

agaffney committed:
Config toggle for duplicate dict key warnings (fixes #56799) (#56805)

bcoca committed:
Preserve ssh error (#56460)

* Preserve ssh error

* more details on fail

* removed redundant caption to errors

samdoran committed:
Ensure uri module always returns status even on failure (#56240)

- Also return url and update docs for other values to indicate they are only returned on success.
- Add integration tests
- Use info variable for common return values
- Use -1 as default status rather than None. This is lines up with with existing code in urls.py
- Add unit tests to ensure status and url are returned on failure

mkrizek committed:
yum: handle "_none_" value for proxy (#56725)

* yum: handle "_none_" value for proxy

Fixes #56538

* Fix sanity check

samccann committed:
Update RHEL repo for Ansible 2.8 (#56859)

samccann committed:
Copyright fix for 2018 (#56860)

* update copyright in footer

* remove extraneous copyright from cli commands

machacekondra committed:
kubevirt: Fix RS and presets template parameters (#56865)

machacekondra committed:
kubevirt_vm: Fix checking of VM update (#56846)

Fred-sun committed:
Modify azure_rm_devtestlabschedule related document (#56819)

webknjaz committed:
📝 Add a GitHub-specific security policy note

PR #56866

It'll show up at
https://github.com/ansible/ansible/security/policy
and will help us guide users to the proper docs page

Co-authored-by: Alicia Cozine <879121+acozine@users.noreply.github.com>

acozine committed:
CLI docs include license not copyright (#56871)

ttencate committed:
Keep machinectl from parsing options of cmd (#56572)

This is needed on machinectl versions prior to v231 (https://github.com/systemd/systemd/pull/3095/commits/d6c3537ea91e33d7c87ffe73851e87d366e2a39d to be precise).

Fixes #56571

bcoca committed:
Always keep subset data for callbacks (#56881)

results were only being preserved during no_log
  fixes #56704

sivel committed:
Validate types before asserting lengths (#56882)

samdoran committed:
Update instructions for running module directly (#56883)

The internals of AnsiballZ changed silghtly

Qalthos committed:
Move eos l3 test IP addresses to documentation ranges (#56726)

dmp1ce committed:
Spelling typo (#56878)

Changed 'creatd' to 'created'

zyv committed:
route53_zone: return existing zone id in check mode (#56702)

* Returns zone ID for existing zone or `null`
* route53_zone: add module unit tests
* route53_zone: add compatibility with Python 2.6 to the unit tests
* route53_zone: address pycodestyle warning (add blank line)

mmazur committed:
kubevirt_vm: simplify previous change + update comments (#56897)

Fred-sun committed:
Update azure_rm_devtestlabvirtualnetwork relate document (#56893)

akamch committed:
Windows FAQ: fix reference (#56909)

bcoca committed:
warn on systemd globs (#56645)

* warn on systemd globs
* added note to docs

  fixes 55476

jonbride committed:
Correct typo in playbooks_intro.rst (#56919)

Typo in code block line: ansible-lint veryify-apache.yml should be
ansible-lint verify-apache.yml

dominikholler committed:
Handle binding:vnic_type as optional (#55526)

According to the OpenStack Networking API
the attribute binding:vnic_type of a port is optional.
This change enables the os_port module to handle
binding:vnic_type as optional.

StingRayZA committed:
Add dns_domain into os_network.py (#56453)

* Add dns_domain into os_network.py

* Added changelog fragment for PR 56453

* fixed linting issue (lib/ansible/modules/cloud/openstack/os_network.py)

pilou- committed:
Handle openstacksdk < 0.10.0: fix AttributeError (#50285)

* openstacksdk < 0.10.0: fix AttributeError

`openstack.version.__version__` expression raises an `AttributeError`
exception when openstacksdk < 0.10.0 is used. `openstack.version` is
now imported as a module, which works for all openstacksdk versions.

Error was:

    The full traceback is:
    Traceback (most recent call last):
      File "$HOME/.ansible/tmp/ansible-tmp-1545612308.8-46792777824159/AnsiballZ_os_security_group.py", line 113, in <module>
        _ansiballz_main()
      File "$HOME/.ansible/tmp/ansible-tmp-1545612308.8-46792777824159/AnsiballZ_os_security_group.py", line 105, in _ansiballz_main
        invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
      File "$HOME/.ansible/tmp/ansible-tmp-1545612308.8-46792777824159/AnsiballZ_os_security_group.py", line 48, in invoke_module
        imp.load_module('__main__', mod, module, MOD_DESC)
      File "/tmp/ansible_os_security_group_payload_keFTIJ/__main__.py", line 163, in <module>
      File "/tmp/ansible_os_security_group_payload_keFTIJ/__main__.py", line 115, in main
      File "/tmp/ansible_os_security_group_payload_keFTIJ/ansible_os_security_group_payload.zip/ansible/module_utils/openstack.py", line 121, in openstack_cloud_from_module
    AttributeError: 'module' object has no attribute 'version'

* Add missing space & dot in error message

* Always require openstacksdk >= 0.12

tayebBenAchour committed:
Update package.py (#56731)

The "-" does not work as expected if I want to get a specific version of a package. After replacing the dash with the equal sign. the task passed successfully.
Depends on package manager - adds example so we show both syntaxes.

bcoca committed:
added tests for log_plays (#56811)

* added tests for log_plays

* grep for warning

* Update tests to work with ansible-test.

wojtek0806 committed:
adds ltm policy to bigip facts (#56926)

facts no longer return changed but queried status in results

mkrizek committed:
Properly detect is_chroot fact on XFS for non-root (#56899)

Fixes #56437

bcoca committed:
Not native, but text (#55676)

* use to_text instead of to_native
* cleaned up some imports and other pyflakisms
* fix missing lib messages

Andersson007 committed:
postgresql: remove depricated lib (#56641)

mkrizek committed:
Migrate away from pipes.quote (#56785)

* Migrate away from pipes.quote

* Fix sanity

mkrizek committed:
ansible-test: prefer shlex.quote (#56823)

bcoca committed:
some logging fixes (#56311)

* some logging fixes

  fixes #25757, #25758, #25761
  alternative to #41859 and #25765

* better color handling, courtesy of alikins

felixfontein committed:
docker conncetion plugin: accept 'dev' as 'newest version' (#56947)

* Accept 'dev' as 'newest version'.

* Add changelog.

* Fix YAML.

SirFerdek committed:
[docker_network] Fix idempotency when using aux_addresses in ipam_config (#56901)

* [docker_network] Fix idempotency when using aux_addresses in ipam_config

Mismatch between keys returned by Docker API (AuxilliaryAddresses) vs
expected by Ansible module (aux_addresses) resulted in tasks always
have status 'changed'. The existing code normalizing one set of
keys to another missed this special case where converting
CamelCase to lowercase is not sufficent.

Please see
https://github.com/moby/moby/blob/master/api/types/network/network.go
for reference.

* Correct keywords formatting in changelog file

Co-Authored-By: Felix Fontein <felix@fontein.de>

ruimoreira committed:
implementing check mode

ruimoreira committed:
implementing check mode

ruimoreira committed:
Merge branch 'dms_replication_subnet_group' of github.com:ruimoreira/ansible into dms_replication_subnet_group

ruimoreira committed:
Merge branch 'dms_replication_subnet_group' of github.com:ruimoreira/ansible into dms_replication_subnet_group
